openedx/core/djangoapps/util/mixins.py

```diff
@@ -0,0 +1,19 @@
+"""Mixins for use during testing."""
+import json
+
+import httpretty
+
+
+class MockApiMixin(object):
+    """
+    Mixin mocking API URLs and providing fake data for testing.
+    """
+    def mock_api(self, url, data, status_code=200, reset_uri=True):
+        """Utility for mocking out API URLs."""
+        self.assertTrue(httpretty.is_enabled(), msg='httpretty must be enabled to mock Programs API calls.')
```

Comment:
Your code would be more readable if you didn't require that the URL in need of mocking be constructed and passed to this method every time it's used. Seeing as how you still require the use of ProgramsDataMixin and CredentialsDataMixin, it'd be cleaner keep this method within those classes. If you want to use a mixin to cut down on some duplication here, you could make the API URL in need of mocking a class attribute and make this method more generic, but I think that introduces unnecessary complexity.